source/server/filter_chain_manager_impl.cc

```diff
@@ -53,13 +53,32 @@ void FilterChainManagerImpl::addFilterChain(
       const auto& cidr_range = Network::Address::CidrRange::create(destination_ip);
       destination_ips.push_back(cidr_range.asString());
     }
-
+    if (destination_ips.empty()) {
+      if (Network::Address::ipFamilySupported(AF_INET6)) {
+        destination_ips.push_back(Network::Utility::getIpv6CidrCatchAllAddress());
+      }
+      // Insert a match-all address. Otherwise if we insert another filter chain with explicit
+      // address would hide the match-all matcher.
+      if (destination_ips.empty() || Network::Address::ipFamilySupported(AF_INET)) {
+        destination_ips.push_back(Network::Utility::getIpv4CidrCatchAllAddress());
+      }
+    }
     std::vector<std::string> source_ips;
     source_ips.reserve(filter_chain_match.source_prefix_ranges().size());
     for (const auto& source_ip : filter_chain_match.source_prefix_ranges()) {
       const auto& cidr_range = Network::Address::CidrRange::create(source_ip);
       source_ips.push_back(cidr_range.asString());
     }
+    if (source_ips.empty()) {
+      if (Network::Address::ipFamilySupported(AF_INET6)) {
+        source_ips.push_back(Network::Utility::getIpv6CidrCatchAllAddress());
+      }
+      // Insert a match-all address. Otherwise if we insert another filter chain with explicit
+      // address would hide the match-all matcher.
+      if (source_ips.empty() || Network::Address::ipFamilySupported(AF_INET)) {
```

Comment:
Question: do we run v4/v6 tests in which this branch gets hit here and below? Do we have a test that covers this case?

/wait-any

Reply:
I cannot imagine a machine running with neither IPv4 nor ipv6...
My machine(and most of the machines) has ipv4 so the branch is definitely hit. 

Let me check if the ipv6-specific tests cover this test case

Reply:
Obviously it's not neither, but I'm not sure we have tests for IPv6 only or dual stack. I would like to make sure we do.

Reply:
OK... ipv6 test doesn't includes this one. Figuring out how to add it